src/main/java/com/github/onsdigital/babbage/api/util/LegacyProxyMaxAge.java

```diff
@@ -0,0 +1,64 @@
+package com.github.onsdigital.babbage.api.util;
+
+import java.util.Enumeration;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import com.github.onsdigital.logging.v2.event.SimpleEvent;
+import static com.github.onsdigital.babbage.configuration.ApplicationConfiguration.appConfig;
+
+public class LegacyProxyMaxAge {
+    public int getMaxAgeValueFromLegacyProxy(HttpServletRequest request) {
+        RequestBuilder requestBuilder = buildRequest(request);          
+        HttpUriRequest forwardRequest = requestBuilder.build();
+
+        int maxAgeValue = 0;
+        try {
+            CloseableHttpClient httpClient = HttpClients.createDefault();
+            CloseableHttpResponse proxyResponse = httpClient.execute(forwardRequest);
+            maxAgeValue = extractMaxAgeValue(proxyResponse);
+        } catch (Exception e) {
+            SimpleEvent.error().exception(e).log("api MaxAge: error getting max age from legacy proxy");
+        }
+        return maxAgeValue;    
+    }
+    
+    private RequestBuilder buildRequest(HttpServletRequest request) {
+        RequestBuilder requestBuilder = RequestBuilder.create(request.getMethod())
+                .setUri(appConfig().babbage().getLegacyCacheProxyUrl());
```

Comment:
You should require the Legacy Cache Proxy URL as a parameter of either the `getMaxAgeValueFromLegacyProxy` function or the constructor. That way, your code is more decoupled (this class won't depend on `appConfig`).